bodo/pandas/managers.py

```diff
@@ -194,6 +198,32 @@ def __repr__(self) -> str:
         else:
             return super().__repr__()
 
+    def execute_plan(self):
+        from bodo.pandas.utils import execute_plan
+
+        data = execute_plan(self._plan)
+        if isinstance(data, BodoLazyWrapper):
+            # We got a lazy result, we need to take ownership of the result
+            # and transfer ownership of the data to this manager
+            self._plan = None
+            self._md_result_id = data._mgr._md_result_id
+            self._md_nrows = data._mgr._md_nrows
+            self._md_head = data._mgr._md_head
+            self._collect_func = data._mgr._collect_func
+            self._del_func = data._mgr._del_func
+            self.axes = data._mgr.axes
+            # Transfer ownership to this manager
+            data._mgr._md_result_id = None
+            out = type(data).from_lazy_mgr(self, self._md_head)
+            return out
```

Comment:
It's a bit confusing that this function returns a new object. Is it basically an object reusing the same manager?

Reply:
I agree.  We've got an object and it's manager but after executing the plan we convert cpp table back to df and in that process we create a new object and new manager.  We are now taking key parts of the new objects and putting them in the old objects.  Wish there was a better way.